internal/policies/mount/mount_test.go

```diff
@@ -0,0 +1,106 @@
+package mount
+
+import (
+	"context"
+	"flag"
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/policies/entry"
+)
+
+func TestApplyPolicy(t *testing.T) {
+	t.Parallel()
+
+	entries := GetEntries(5, 0, 1, 0, []string{"\n"})
+
+	tests := map[string]struct {
+		entries    []entry.Entry
+		objectName string
+		computer   bool
+		uid        string
+		gid        string
+
+		wantErr bool
+	}{
+		"successfully generates mounts file":        {entries: entries},
+		"generates no file if there are no entries": {entries: []entry.Entry{}},
+
+		"error when user is not found":    {entries: entries, objectName: "dont exist", wantErr: true},
+		"error when user has invalid uid": {entries: entries, uid: "invalid", wantErr: true},
+		"error when user has invalid gid": {entries: entries, gid: "invalid", wantErr: true},
+
+		// To be removed when computer policies get implemented.
+		"error when trying to apply computer policies": {entries: entries, computer: true, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			mountsPath := filepath.Join(t.TempDir(), t.Name(), "mounts")
+
+			userServicePath := filepath.Join("testdata", "adsys-user-mount.service")
+
+			opts := []Option{WithMountsFilePath(mountsPath), WithUserMountServicePath(userServicePath)}
+
+			usr, _ := user.Current()
+			if tc.objectName == "" {
+				if tc.uid == "" {
+					tc.uid = usr.Uid
+				}
+				if tc.gid == "" {
+					tc.gid = usr.Gid
+				}
+
+				tc.objectName = "ubuntu"
+
+				opts = append(opts, WithUserLookup(func(string) (*user.User, error) {
+					return &user.User{Uid: tc.uid, Gid: tc.gid}, nil
+				}))
+			}
+
+			m := New(opts...)
+
+			err := m.ApplyPolicy(context.Background(), tc.objectName, tc.computer, tc.entries)
+			if tc.wantErr {
+				require.Error(t, err, "Expected an error but got none")
+				return
+			}
+			require.NoError(t, err, "Expected no error but got one")
+
+			b, err := os.ReadFile(mountsPath)
+			if len(tc.entries) == 0 {
+				require.Error(t, err, "Expected no mounts file")
+				return
+			}
+			require.NoError(t, err, "Expected to read generated mounts file")
+
+			testdata := filepath.Join("testdata", t.Name())
+			wantPath := filepath.Join(testdata, "mounts")
+			if update {
+				err = os.MkdirAll(testdata, 0750)
```

Comment:
Never write directly to testdata please:
- Here, you don’t clean old files (so any files that we remove in the future during the golden generation will still stay).
- If there is a middle in the middle of the refresh, you end up with mixed parsed data.

Also, there are some functions in testutils (again, look at the script manager to get inspiration), using `CompareTreesWithFiltering` for instance. This allow you then to ensure you keep and tests properly the whole hierarchy that the manager should creates in addition to the file itself.